app/models/log_event.rb

```diff
@@ -3,8 +3,44 @@ class LogEvent < ActiveRecord::Base
   belongs_to :user
   belongs_to :loggable, polymorphic: true
 
-  def self.log(loggable, event_type, user)
-    create(loggable: loggable, event_type: event_type, user_id: user.id)
+  def self.log(loggable, event_type, event_time, user)
+    create(
+      loggable: loggable, event_type: event_type,
+      event_time: event_time, user_id: user.try(:id))
+  end
+
+  def self.search(start_date: nil, end_date: nil, events: [])
+    result = all
+    result = result.filter_date(start_date, end_date) if start_date || end_date
+    result = result.filter_event(events) if events.present?
+    result
+  end
+
+  def self.filter_date(start_date, end_date)
+    dates = [(start_date || 10.years.ago), (end_date || Date.current)]
```

Comment:
Will this filter to the end of day for `end_date`? If I search for 7/11-7/24, we should include 7/24. That keeps it consistent with other searches in nucore.